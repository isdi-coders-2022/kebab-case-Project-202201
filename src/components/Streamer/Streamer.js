import styled from "styled-components";
import PropTypes from "prop-types";
import ButtonImage from "../ButtonImage/ButtonImage";
import useFavoritesAPI from "../../hooks/useFavoritesAPI";
import { useNavigate } from "react-router-dom";

const ArticleCard = styled.article`
  width: calc(100% - 40px);
  margin: 10px;
  height: 121px;
  left: 15px;
  top: 15px;
  background: #efeff1;
  display: flex;
  flex-direction: row;
`;

const ColumnCard1 = styled.div``;

const ColumnCard2 = styled.div`
  padding-left: 15px;
  display: flex;
  flex-direction: column;
  padding-right: 10px;
`;

const RoundedImage = styled.img`
  position: relative;
  width: 100px;
  height: 100px;
  border-radius: 50px;
  left: 10px;
  top: 10px;
`;

const BroadcasterType = styled.p`
  position: relative;
  width: 70px;
  height: 18px;
  left: 25px;
  bottom: 10px;

  font-family: Roboto;
  font-style: normal;
  font-weight: 800;
  font-size: 14px;
  line-height: 16px;

  text-align: center;
  letter-spacing: 0.4px;
  background-color: #5d16c6;
  color: #ffffff;
`;

const NameStreamer = styled.p`
  padding-top: 10px;
  padding-right: 10px;
  text-align: right;
  font-family: Roboto;
  font-style: normal;
  font-weight: 800;
  font-size: 24px;
  line-height: 16px;

  letter-spacing: 0.4px;

  color: #5d16c6;
`;

const OnlineType = styled.p`
  padding-right: 10px;
  text-align: right;
  font-family: Roboto;
  font-style: normal;
  font-weight: 800;
  font-size: 14px;
  line-height: 16px;

  text-align: right;
  letter-spacing: 0.4px;

  color: #0fa60c;
`;

const StreamerDescription = styled.p`
  padding-top: 10px;
  padding-right: 10px;
  text-align: right;

  font-family: Roboto;
  font-style: normal;
  font-weight: 800;
  font-size: 14px;
  line-height: 16px;
  letter-spacing: 0.4px;

  color: #000000;
`;

const Streamer = ({ streamerInfo }) => {
  const info = streamerInfo;
  const { sendStreamer } = useFavoritesAPI();
  const addToFavs = () => {
    const newStreamer = {
      id: info.id,
      user_id: info.user_id,
      display_name: info.display_name,
      profile_image_url: info.profile_image_url,
    };
    sendStreamer(newStreamer);
  };

  let navigate = useNavigate();
  const viewDetails = () => {
    navigate(`/info/${streamerInfo[0].id}`);
  };

  return (
    streamerInfo && (
      <ArticleCard className="streamerMain" onClick={viewDetails}>
        <ColumnCard1>
          <RoundedImage
            className="streamerPicture"
            alt="Streamer"
            src={info.profile_image_url}
          ></RoundedImage>
          <BroadcasterType className="broadcasterType">
            {info.broadcaster_type}
          </BroadcasterType>
        </ColumnCard1>
        <ColumnCard2>
          <NameStreamer className="displayName">
            {`${info.display_name}`}
          </NameStreamer>
          <OnlineType className="onlineType">{info.type}</OnlineType>

          <StreamerDescription className="StreamerDescription">
            {info.description.length > 100
              ? info.description.slice(0, 99) + "..."
              : info.description}
          </StreamerDescription>
        </ColumnCard2>
        <ButtonImage
          favClass={"newClass"}
          image={"img/favDown.png"}
          imageAlt={"shawarma icon"}
          actionOnClick={addToFavs}
        />
      </ArticleCard>
    )
  );
};

Streamer.propTypes = {
  id: PropTypes.number,
  display_name: PropTypes.string,
  broadcaster_type: PropTypes.string,
  description: PropTypes.string,
  profile_image_url: PropTypes.string,
  actionClick: PropTypes.func,
};

export default Streamer;
