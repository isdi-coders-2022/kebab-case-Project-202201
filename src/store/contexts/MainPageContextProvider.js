import MainPageContext from "./MainPageContext";

const MainPageContextProvider = ({ children }) => {
  return <MainPageContext.Provider>{children}</MainPageContext.Provider>;
};
