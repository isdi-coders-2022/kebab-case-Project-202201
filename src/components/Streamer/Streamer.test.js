import Streamer from "./Streamer";
import { render, screen } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import MainPageContextProvider from "../../store/contexts/MainPageContextProvider";

describe("Given the Streamer component", () => {
  const streamerInfo2 = {
    id: "124422593",
    display_name: "LEC",
    type: "online",
    broadcaster_type: "partner",
    description:
      "Official Twitch Channel of the League of Legends European Championship. For more information, schedules and stats head to Lolesports.com",
    profile_image_url:
      "https://static-cdn.jtvnw.net/jtv_user_pictures/41b23352-5c99-4eb2-974b-1e9b8120425e-profile_image-300x300.png",
  };

  describe("When it's rendered", () => {
    test("Then it should show 'partner' in the document", () => {
      render(
        <BrowserRouter>
          <MainPageContextProvider>
            <Streamer streamerInfo={streamerInfo2}></Streamer>
          </MainPageContextProvider>
        </BrowserRouter>
      );

      const streamerElement = screen.getByText("LEC");
      expect(streamerElement).toBeInTheDocument();
    });

    test("Then it should show 'LEC' in the document", () => {
      render(
        <BrowserRouter>
          <MainPageContextProvider>
            <Streamer streamerInfo={streamerInfo2}></Streamer>
          </MainPageContextProvider>
        </BrowserRouter>
      );

      const streamerElement = screen.getByText("LEC");
      expect(streamerElement).toBeInTheDocument();
    });
  });

  describe("When it's rendered with a description shorter than 100", () => {
    test("Then it should show 'texto breve' in the document", () => {
      streamerInfo2.description = "texto breve";
      render(
        <BrowserRouter>
          <MainPageContextProvider>
            <Streamer streamerInfo={streamerInfo2}></Streamer>
          </MainPageContextProvider>
        </BrowserRouter>
      );

      const streamerElement = screen.getByText(/texto breve/);
      expect(streamerElement).toBeInTheDocument();
    });
  });

  describe("When it gets a const streamer that creates a streamer with StreamerInfo as props", () => {
    test("then it should create and compare a snapshot with Streamer with StreamerInfo as component", () => {
      const streamer = TestRenderer.create(
        <BrowserRouter>
          <MainPageContextProvider>
            <Streamer streamerInfo={streamerInfo2} />
          </MainPageContextProvider>
        </BrowserRouter>
      );
      expect(streamer.toJSON()).toMatchSnapshot();
    });
  });
});
