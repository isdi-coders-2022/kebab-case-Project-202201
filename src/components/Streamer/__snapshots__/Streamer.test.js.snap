// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the Streamer component When it gets a const streamer that creates a streamer with StreamerInfo as props then it should create and compare a snapshot with Streamer with StreamerInfo as component 1`] = `
<article
  className="sc-gsDKAQ gjFBph streamerMain"
  onClick={[Function]}
>
  <div
    className="sc-dkPtRN"
  >
    <img
      alt="Streamer"
      className="sc-eCImPb jtrcDZ streamerPicture"
      src="https://static-cdn.jtvnw.net/jtv_user_pictures/41b23352-5c99-4eb2-974b-1e9b8120425e-profile_image-300x300.png"
    />
    <p
      className="sc-jRQBWg cFTUEa broadcasterType"
    >
      partner
    </p>
  </div>
  <div
    className="sc-hKwDye kYObYN"
  >
    <p
      className="sc-gKclnd bEvnvO displayName"
    >
      LEC
    </p>
    <p
      className="sc-iCfMLu bCfYNq onlineType"
    >
      online
    </p>
    <button
      className="sc-bdvvtL jAUNTG"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="shawarma icon"
        src="img/multidurum.png"
      />
    </button>
    <p
      className="sc-furwcr kDLFXU StreamerDescription"
    >
      texto breve
    </p>
  </div>
</article>
`;
